lib/trino-filesystem-gcs/src/main/java/io/trino/filesystem/gcs/GcsFileSystemConfig.java

```diff
@@ -0,0 +1,161 @@
+    private String projectId;
+
+    private boolean useGcsAccessToken;
+    private String jsonKey;
+    private String jsonKeyFilePath;
+
+    @NotNull
+    public DataSize getReadBlockSize()
+    {
+        return readBlockSize;
+    }
+
+    @Config("gcs.read-block-size")
+    @ConfigDescription("Minimum size that will be read in one rpc. The default size is 2MiB, see com.google.cloud.BaseStorageReadChannel.")
+    public GcsFileSystemConfig setReadBlockSize(DataSize readBlockSize)
+    {
+        this.readBlockSize = readBlockSize;
+        return this;
+    }
+
+    @NotNull
+    public DataSize getWriteBlockSize()
+    {
+        return writeBlockSize;
+    }
+
+    @Config("gcs.write-block-size")
+    @ConfigDescription("Minimum size that will be written in one rpc. The default size is 16MiB, see com.google.cloud.BaseStorageWriteChannel.")
+    public GcsFileSystemConfig setWriteBlockSize(DataSize writeBlockSize)
+    {
+        this.writeBlockSize = writeBlockSize;
+        return this;
+    }
+
+    @Min(1)
+    public int getPageSize()
+    {
+        return pageSize;
+    }
+
+    @Config("gcs.page-size")
+    @ConfigDescription("The maximum number of blobs to return per page.")
+    public GcsFileSystemConfig setPageSize(int pageSize)
+    {
+        this.pageSize = pageSize;
+        return this;
+    }
+
+    @Min(1)
+    public int getBatchSize()
+    {
+        return batchSize;
+    }
+
+    @Config("gcs.batch-size")
+    @ConfigDescription("Number of blobs to delete per batch. Recommended batch size is 100: https://cloud.google.com/storage/docs/batch.")
+    public GcsFileSystemConfig setBatchSize(int batchSize)
+    {
+        this.batchSize = batchSize;
+        return this;
+    }
+
+    @Nullable
+    public String getProjectId()
+    {
+        return projectId;
+    }
+
+    @Config("gcs.projectId")
+    public GcsFileSystemConfig setProjectId(String projectId)
+    {
+        this.projectId = projectId;
+        return this;
+    }
+
+    public boolean isUseGcsAccessToken()
+    {
+        return useGcsAccessToken;
+    }
+
+    @Config("gcs.use-access-token")
+    public GcsFileSystemConfig setUseGcsAccessToken(boolean useGcsAccessToken)
+    {
+        this.useGcsAccessToken = useGcsAccessToken;
+        return this;
+    }
+
+    @Nullable
+    public String getJsonKey()
+    {
+        return jsonKey;
+    }
+
+    @Config("gcs.json-key")
+    @ConfigSecuritySensitive
+    public GcsFileSystemConfig setJsonKey(String jsonKey)
+    {
+        this.jsonKey = jsonKey;
+        return this;
+    }
+
+    @Nullable
+    @FileExists
+    public String getJsonKeyFilePath()
+    {
+        return jsonKeyFilePath;
+    }
+
+    @Config("gcs.json-key-file-path")
+    @ConfigDescription("JSON key file used to access Google Cloud Storage")
+    public GcsFileSystemConfig setJsonKeyFilePath(String jsonKeyFilePath)
+    {
+        this.jsonKeyFilePath = jsonKeyFilePath;
+        return this;
+    }
+
+    public void validate()
```

Comment:
Can you use `@AssertTrue` here? See `SqlEnvironmentConfig` for an example

Reply:
Tried again, problem is that `@AssertTrue` fails in `testExplicitPropertyMappings` since one of `jsonKey` or `jsonKeyFilePath` must be null but you can't specify a null in the`properties` map in the test.